x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -95,133 +202,158 @@ export const update = ({
     return Object.keys(updateCaseAttributes).length > 0;
   });
 
-  if (updateFilterCases.length > 0) {
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    const { username, full_name, email } = await caseService.getUser({ request });
-    const updatedDt = new Date().toISOString();
-    const updatedCases = await caseService.patchCases({
-      client: savedObjectsClient,
-      cases: updateFilterCases.map((thisCase) => {
-        const { id: caseId, version, ...updateCaseAttributes } = thisCase;
-        let closedInfo = {};
-        if (updateCaseAttributes.status && updateCaseAttributes.status === CaseStatuses.closed) {
-          closedInfo = {
-            closed_at: updatedDt,
-            closed_by: { email, full_name, username },
-          };
-        } else if (
-          updateCaseAttributes.status &&
-          (updateCaseAttributes.status === CaseStatuses.open ||
-            updateCaseAttributes.status === CaseStatuses['in-progress'])
-        ) {
-          closedInfo = {
-            closed_at: null,
-            closed_by: null,
-          };
-        }
-        return {
-          caseId,
-          updatedAttributes: {
-            ...updateCaseAttributes,
-            ...closedInfo,
-            updated_at: updatedDt,
-            updated_by: { email, full_name, username },
-          },
-          version,
-        };
-      }),
-    });
+  if (updateFilterCases.length <= 0) {
+    throw Boom.notAcceptable('All update fields are identical to current version.');
+  }
 
-    // If a status update occurred and the case is synced then we need to update all alerts' status
-    // attached to the case to the new status.
-    const casesWithStatusChangedAndSynced = updateFilterCases.filter((caseToUpdate) => {
-      const currentCase = myCases.saved_objects.find((c) => c.id === caseToUpdate.id);
-      return (
-        currentCase != null &&
-        caseToUpdate.status != null &&
-        currentCase.attributes.status !== caseToUpdate.status &&
-        currentCase.attributes.settings.syncAlerts
-      );
-    });
+  const casesMap = myCases.saved_objects.reduce((acc, so) => {
```

Comment:
As you create a map, it would be nice to replace all `myCases.saved_objects.find(...)` in the file with `caseMap.get(...)` to avoid searching. For example when searching for `conflictedCases` and `updateCases` above.

Reply:
Yeah great idea. If it's ok, I'll probably do that in a follow up refactoring PR.